server/route.go

```diff
@@ -410,6 +415,20 @@ func (c *client) processRouteInfo(info *Info) {
 		return
 	}
 
+	// Detect if we have a mismatch of cluster names.
+	if info.Cluster != "" && info.Cluster != clusterName {
+		c.mu.Unlock()
+		// If we are dynamic we may update our cluster name.
+		if c.srv.getOpts().Cluster.Name == "" && strings.Compare(clusterName, info.Cluster) < 0 {
```

Comment:
Yes, but if the remote is not dynamic (and this one is), should we accept? It seem strange that we accept only if it is dynamic and remote name is >.

Reply:
We can't override non-dynamic names. That is cluster config error.

Reply:
Fixed.